apps/settings/js/modules/sim_security.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * This is a simple model that can help us mimic the status change event
+ * when enabling / disabling simcard.
+ *
+ * @module SimSecurity
+ */
+define(function(require) {
+  'use strict';
+
+  var Module = require('modules/base/module');
+  var SettingsUtils = require('modules/settings_utils');
+  var EventEmitter = require('modules/base/event_emitter');
+
+  var APIS = [
+    'unlockCardLock',
+    'setCardLock',
+    'updateContact',
+    'getCardLock',
+    'getCardLockRetryCount'
+  ];
+
+  var EVENT_CHANGE_APIS = [
+    'unlockCardLock',
+    'setCardLock'
+  ];
+
+  var SimSecurity = Module.create(function() {
+    this.super(EventEmitter).call(this, [
+      'sim-security-pin-enabled',
+      'sim-security-pin-disabled',
+      'sim-security-pin2-enabled',
+      'sim-security-pin2-disabled',
+      'sim-security-fdn-enabled',
+      'sim-security-fdn-disabled',
+      'sim-security-pin-changed',
+      'sim-security-pin2-changed'
+    ]);
+  }).extend(EventEmitter);
+
+  APIS.forEach(function(apiName) {
+    SimSecurity.prototype[apiName] = function(cardIndex) {
+      if (arguments.length < 2) {
+        this.throw(
+          'You are using SimSecurity API wrongly and you may forget' +
+          'to put your cardIndex, please check again');
+      }
+
+      var passedArguments = [].slice.call(arguments, 1);
+      var icc = SettingsUtils.getIccByCardIndex(cardIndex);
```

Comment:
I still think this is a general function that would be used by the others. I can make this function as its internal function first and see what to do in the future.